crates/iceberg/src/metadata_scan.rs

```diff
@@ -0,0 +1,265 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Metadata table api.
+
+use std::sync::Arc;
+
+use arrow_array::builder::{MapBuilder, PrimitiveBuilder, StringBuilder};
+use arrow_array::types::{Int64Type, TimestampMillisecondType};
+use arrow_array::RecordBatch;
+use arrow_schema::{DataType, Field, Schema, TimeUnit};
+
+use crate::spec::TableMetadataRef;
+use crate::table::Table;
+use crate::Result;
+
+/// Table metadata scan.
+///
+/// Used to inspect a table's history, snapshots, and other metadata as a table.
+///
+/// See also <https://iceberg.apache.org/docs/latest/spark-queries/#inspecting-tables>.
+#[derive(Debug)]
+pub struct MetadataScan {
+    metadata_ref: TableMetadataRef,
+}
```

Comment:
Wondering what you think about

```suggestion
pub struct MetadataScan<'a> {
    table: &'a Table,
}
```

so we can get at `FileIO`.